### PR TITLE
Modify 'Contact WooCommerce Support' badge in Payments > Settings page

### LINKED ISSUE #6839: Evaluate Additional Merchant Details for BNPLs per Supported Country
### Description

We need to evaluate additional merchant details requirements for Buy Now, Pay Later (BNPL) payment methods for each country we support. This evaluation is necessary to better understand regional requirements and implement appropriate features.

### Acceptance criteria

* Identify any additional merchant details we need to collect for each country and payment method
* Collaborate with Adam on creating documentation to highlight why new PMs might be disabled

### Additional context

Check this Slack thread: p1689364005427329-slack-C04VCT34PCP

### PR DESCRIPTION
**Pull request by mgascam**
Fixes #6839 

#### Changes proposed in this Pull Request

This pull request introduces modifications to the 'Contact WooCommerce Support' badge located on the 'Payments accepted on checkout' section of Payments > Settings.

Screenshot taken on 2023-07-14 at 19 46 31 UTC@2x

This badge is visible when a payment method cannot be enabled because the merchant needs to provide additional details to Stripe. Since merchants can self-serve this missing details through their connected account dashboard in Stripe, we are modifying the text to inform them that there's missing information, including a link to the public documentation that explains the steps to take. 

Screenshot 2023-07-28 at 18 17 19

In addition to the functional changes mentioned earlier, I'm also taking this opportunity to migrate the involved files to TypeScript (TS). 

#### Testing instructions

* You'll need a Stripe account in a country that requires additional information, for instance Canada.
* Enable BNPLs payment method for the store (feature flag and fees).
* Make sure the UPE payment methods are enabled in the store.
* Navigate to WC Admin > Payments > Settings.
* Check that the new badge appears next to the Afterpay payment method.
* Check that a tooltip appear when you hover the badge.
* Check that the 'Learn more' link in the tooltip takes you to https://woocommerce.com/document/woopayments/payment-methods/buy-now-pay-later/#contact-support.
* Disable UPE payment methods
* Click in 'Enable in your store' button to re-enable them.
* In the Additional payment methods screen, check that the new badge appears and the link takes you to the same documentation page
-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6875` or branch name `update/6839-modify-contact-support-badge` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=update/6839-modify-contact-support-badge) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 849fc4f590bc9c1460a4ea1da9f0235c4c95b43b
- Build time: 2023-08-02 11:35:44 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/payment-methods-checkboxes/payment-method-checkbox.tsx [hunk 1]
```diff
@@ -154,17 +169,31 @@ const PaymentMethodCheckbox = ( {
 					) }
 					{ disabled && (
 						<HoverTooltip
-							content={ sprintf(
-								__(
-									'To use %s, please contact WooCommerce support.',
+							content={ interpolateComponents( {
+								//TODO: translators hint.
```
Comment (KarlisJ) [inline client/components/payment-methods-checkboxes/payment-method-checkbox.tsx]: TODO left

**Review by KarlisJ — COMMENTED** [review]
Code review only. 

Overall, I like the changes. One suggestion. To me it seems that both tooltips are defined exactly the same. Is there a reason you haven't extracted it as separate component to make it clear that they are the same?

File: client/components/payment-methods-checkboxes/payment-method-checkbox.tsx (hunk 1 above)
Comment (mgascam) [reply client/components/payment-methods-checkboxes/payment-method-checkbox.tsx]: Done [here](https://github.com/Automattic/woocommerce-payments/pull/6875/commits/0ac879f7b8c8064343af0b86864bc5f5cd445100) thanks!

**Review by dmvrtx — APPROVED** [review]
Code-wise looks good. I've tried to trigger condition for the badge by by overriding properties, but my React-fu failed me.

**Comment by KarlisJ** [comment]
I'll do a test tomorrow

File: client/components/payment-methods-checkboxes/payment-method-checkbox.tsx [hunk 2]
```diff
@@ -154,17 +169,31 @@ const PaymentMethodCheckbox = ( {
...
-								PaymentMethodsMap[ name ].label
-							) }
+								components: {
+									learnMoreLink: (
+										// eslint-disable-next-line jsx-a11y/anchor-has-content
+										<a
+											target="_blank"
+											rel="noreferrer"
+											/* eslint-disable-next-line max-len */
+											href={ getDocumentationUrlForDisabledPaymentMethod(
```
Comment (kalessil) [inline client/components/payment-methods-checkboxes/payment-method-checkbox.tsx]: Given we'll be re-working this, this change should be easy to remove. The current approach is not that centralized for being easy to remove.

I'd suggest introducing a new component extending `HoverTooltip` and encapsulating the logic, constants and translations injection - all inside.

**Review by kalessil — CHANGES_REQUESTED** [review]
It's a lot of great changes here in this PR. I'm leaving a comment, which shall make the PR even better.

File: client/components/payment-methods-checkboxes/payment-method-checkbox.tsx (hunk 2 above)
Comment (KarlisJ) [reply client/components/payment-methods-checkboxes/payment-method-checkbox.tsx]: I'm with Vlad here, that's a good suggestion

**Comment by KarlisJ** [comment]
Tested with CA merchant locally. I see the badge as expected.

File: client/components/payment-methods-checkboxes/payment-method-checkbox.tsx (hunk 2 above)
Comment (mgascam) [reply client/components/payment-methods-checkboxes/payment-method-checkbox.tsx]: Thanks for the suggestion @kalessil @KarlisJ  here's my attempt to implement it [e9661fa](https://github.com/Automattic/woocommerce-payments/pull/6875/commits/e9661fa010508a873ef4fd1ee91fd8c1afa0545f)

File: client/components/payment-method-disabled-tooltip/index.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+						// eslint-disable-next-line jsx-a11y/anchor-has-content
+						<a
+							target="_blank"
+							rel="noreferrer"
+							title={ __(
+								'Learn more about enabling payment methods',
+								'woocommerce-payments'
+							) }
+							/* eslint-disable-next-line max-len */
+							href={ getDocumentationUrlForDisabledPaymentMethod(
```
Comment (kalessil) [inline client/components/payment-method-disabled-tooltip/index.tsx:64]: Looks like the right direction.

Now remaining:
- drop the new utils
- relocate logic and URLs from `getDocumentationUrlForDisabledPaymentMethod` inside this components

**Review by kalessil — CHANGES_REQUESTED** [review]

File: client/components/payment-method-disabled-tooltip/index.tsx (hunk 3 above)
Comment (KarlisJ) [reply client/components/payment-method-disabled-tooltip/index.tsx:64]: Nothing to add here, I concur with Vlad again.

Comment (mgascam) [reply client/components/payment-method-disabled-tooltip/index.tsx:64]: Done here [bf7ef58](https://github.com/Automattic/woocommerce-payments/pull/6875/commits/bf7ef5860273bd235271e8f98cb298236fa39136) thanks!

File: client/components/payment-method-disabled-tooltip/index.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	DEFAULT:
+		'https://woocommerce.com/document/woopayments/payment-methods/additional-payment-methods/#method-cant-be-enabled',
+	BNPLS:
+		'https://woocommerce.com/document/woopayments/payment-methods/buy-now-pay-later/#contact-support',
+};
+
+export const getDocumentationUrlForDisabledPaymentMethod = (
+	paymentMethodId: string
+): string => {
+	let link;
```
Comment (kalessil) [inline client/components/payment-method-disabled-tooltip/index.tsx]: A bit misleading name, what about using `url`?

**Review by kalessil — CHANGES_REQUESTED** [review]
Looks great overall! 

One thing I discovered is that on the setting page, the styles changes are not applied for my setup:
Screenshot from 2023-08-02 09-44-30

Not sure why.

File: client/components/payment-method-disabled-tooltip/index.tsx (hunk 4 above)
Comment (mgascam) [reply client/components/payment-method-disabled-tooltip/index.tsx]: Done here [eae4564](https://github.com/Automattic/woocommerce-payments/pull/6875/commits/eae4564e0bbed5399435a082d414245b4741cfb2) thanks!

**Comment by mgascam** [comment]
> Looks great overall!
> 
> One thing I discovered is that on the setting page, the styles changes are not applied for my setup: Screenshot from 2023-08-02 09-44-30
> 
> Not sure why.

Thanks for checking it @kalessil. I tried with major browsers (Safari, Chrome, Firefox) to discard a compatibility issue, and it is displayed with the correct styles. 
I also checked that the generated CSS contains the correct styles after generating the release build. Not sure if restarting the local build (i.e. npm start) would help you here.  
Also sharing a screenshot of the expected CSS styles as seen in dev tools, I guess you get something different?

Screenshot 2023-08-02 at 13 37 42

**Review by kalessil — APPROVED** [review]
It works now - probably it was some caching issue (apparently, Ctrl+R does not always work as I thought it should).

**Pull request merged**